src/object.rs

```diff
@@ -246,6 +255,49 @@ impl Object {
       scope.to_local(ptr).unwrap()
     }
   }
+
+  fn resolve_js_array<'sc>(
+    &self,
+    scope: &mut impl ToLocal<'sc>,
+    context: Local<Context>,
+    raw_name_ptr: *mut Value,
+  ) -> Option<Vec<String>> {
+    let raw_names: Local<Array> =
+      unsafe { scope.to_local(raw_name_ptr) }?.try_into().ok()?;
+    let mut names: Vec<String> = vec![];
+    for i in 0..raw_names.length() {
+      let local = raw_names.get_index(scope, context, i)?;
+      let local: Local<crate::String> = local.try_into().ok()?;
+      names.push(local.to_rust_string_lossy(scope));
+    }
+    Some(names)
+  }
+
+  /// This function has the same functionality as GetPropertyNames
+  /// but the returned array doesn't contain the names of
+  /// properties from prototype objects.
+  pub fn get_own_property_names<'sc>(
+    &self,
+    scope: &mut impl ToLocal<'sc>,
+    context: Local<Context>,
+  ) -> Option<Vec<String>> {
+    let raw_name_ptr =
+      unsafe { v8__Object__GetOwnPropertyNames(self, context) };
+    self.resolve_js_array(scope, context, raw_name_ptr)
```

Comment:
Why does the returned value need to be processed by `resolved_js_array()`
I'm in favor of adding this binding, but we should just return the `v8::Local<v8::Array>` that v8 gives us.
(Cf. goal #1 as stated in the readme: "the API should match the original API as closely as possible")

Reply:
Gotcha, just fit my needs at that exact moment. Will fixup in a bit.